### PR TITLE
cleanup: join push and push_branch helpers

### PR DESCRIPTION
**Pull request by charlierudolph**
@kevgo @allewun

### CODE REVIEW

File: src/git-extract [hunk 1]
```diff
@@ -59,10 +59,10 @@ function steps {
 
   echo "checkout_main_branch"
   echo "rebase_tracking_branch"
-  echo_if_true "push" "$HAS_REMOTE"
+  echo_if_true "push_branch $MAIN_BRANCH_NAME" "$HAS_REMOTE"
```
Comment (charlierudolph) [inline src/git-extract]: Tried converting this and the two steps above to `sync_branch_steps` but ran into a very interesting issue which I want to deal with separately.

File: src/helpers/git_helpers/branch_helpers.sh [hunk 2]
```diff
@@ -1,6 +1,18 @@
 #!/usr/bin/env bash
 
 
+# Returns whether the given branch is in sync with its tracking branch
```
Comment (kevgo) [inline src/helpers/git_helpers/branch_helpers.sh]: Cool, could we use this function to omit pushes if they are not necessary?

**Comment by kevgo** [comment]
Nice cleanup! :ship:

File: src/helpers/git_helpers/branch_helpers.sh (hunk 2 above)
Comment (charlierudolph) [inline src/helpers/git_helpers/branch_helpers.sh]: We already do!

**Pull request closed without merging**